### ISSUE TITLE
Why did you use "softmax" and "categorical_crossentropy"?

### ISSUE THREAD

**Issue by Kevin-Moon** [orobix/retina-unet#45]
Thank you for your dedicated work.

I want to ask why you used "softmax" and "categorical_crossentropy".

I understand this problem as "Multi Label classification", so usually we use
"sigmoid" and "binary cross entropy".

Is not this multi-label problem?

In short, what does the final layer(conv6 = core.Reshape((2,patch_height*patch_width))(conv6))
mean? especailly I wonder about "2".

thank you for any advices in advance.

**Comment by hungbie** [comment]
What you are refering to is a Binary Classification problem and hence using sigmoid with binary cross entropy is possible. Thus if you are using that you only need to have 1 class in the final prediction.

However here it is indeed still a binary problem but they are using categorical cross entropy and is compatible with more than 2 classes. And since categorical cross entropy is used they need to specify that there are 2 classes, hence 2.

**Comment by Kevin-Moon** [comment]
Thank you for your kind response.

Then, do you think that they have almost the same result in terms of
accuracy or performance?

Thanks in advance.

Best
Kevin

2018. 2. 12. 오후 6:22에 "Pham Tan Hung" <notifications@github.com>님이 작성:

**Comment by hungbie** [comment]
For this problem yes they should be the same. However say we now want to also detect the ONH and have 3 classes instead then we can no longer use Sigmoid with Binary Cross Entropy.

**Comment by Kevin-Moon** [comment]
I got it!!

Thank you so much.

I hope that you will achieve great work.


Best
Kevin

2018. 2. 12. 오후 7:13에 "Pham Tan Hung" <notifications@github.com>님이 작성:
